File: RewardedInterstitialExample/ObjcImplementation/ObjcImplementation/ViewController.m
//
//  ViewController.m
//  ObjcImplementation
//
//  Created by Rishabh Tripathi on 11/06/25.
//

#import "ViewController.h"

@interface ViewController () <ApRewardedInterstitialListener>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

-(IBAction)showAd:(UIButton *)sender {
    [ApAppSDK showRewardedInterstitialAdWithViewController:self apPlacementID:@"testPlacement" apRewardedInterstitialListener:self];
}


- (void)onComplete { 
    NSLog(@"Ad Complete");
}

@end

File: RewardedInterstitialExample/SwiftImplementation/SwiftImplementation/ViewController.swift
//
//  ViewController.swift
//  SwiftImplementation
//
//  Created by Rishabh Tripathi on 09/06/25.
//

import UIKit
import ApMobileSDK

class ViewController: UIViewController, ApRewardedInterstitialListener {
    
    func onComplete() {
        print("Ad Complete")
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }

    @IBAction func showAd(_ sender: UIButton) {
        ApAppSDK.showRewardedInterstitialAd(viewController: self, apPlacementID: "testPlacement", apRewardedInterstitialListener: self)
    }
    
}

